README.md

```diff
@@ -1,19 +1,50 @@
-## WARNING: EXPERIMENTAL
+# WARNING: EXPERIMENTAL
 
 This plugin is currently in an experimental state and not for use on production sites.
 
-<!--
+## Plugin API
```

Comment:
I'm not sure the readme is the right place for the API documentation, maybe we should add a separate PLUGIN-API file?

Reply:
My long term goal is to write an automation to sync a readme with a discourse meta topic, so having all in one file would help :p